vscode-dotnet-runtime-library/src/Utils/WebRequestWorkerSingleton.ts

```diff
@@ -56,100 +56,68 @@ export class WebRequestWorkerSingleton
+            axiosRetry(uncachedAxiosClient, {
+                // Inject a custom retry delay to exponentially increase the time until we retry.
+                retryDelay(retryCount: number)
+                {
+                    return Math.pow(2, retryCount); // Takes in the int as (ms) to delay.
+                }
+            });
 
-        // Record when the web requests begin:
-        // Register this so it happens before the cache https://axios-cache-interceptor.js.org/guide/interceptors#explanation
-        uncachedAxiosClient.interceptors.request.use(config =>
-        {
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-            (config as any).startTime = process.hrtime.bigint();
-            return config;
-        });
+            // Record when the web requests begin:
+            // Register this so it happens before the cache https://axios-cache-interceptor.js.org/guide/interceptors#explanation
+            uncachedAxiosClient.interceptors.request.use(config =>
+            {
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                (config as any).startTime = process.hrtime.bigint();
+                return config;
+            });
 
-        // Record when the server responds:
-        // Register this so it happens after the cache -- this means we need to check if the result is cached before reporting perf data!
-        // ^ the request should not be processed if it is in the cache, it seems nonsensical to check this before the cache is hit even though this is possible
-        uncachedAxiosClient.interceptors.response.use(res =>
-        {
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-            (res as any).startTime = (res.config as any).startTime;
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-            (res as any).finalTime = process.hrtime.bigint();
-            return res;
-        },
-            res => // triggered when status code is not 2XX
+            // Record when the server responds:
+            // Register this so it happens after the cache -- this means we need to check if the result is cached before reporting perf data!
+            // ^ the request should not be processed if it is in the cache, it seems nonsensical to check this before the cache is hit even though this is possible
+            uncachedAxiosClient.interceptors.response.use(res =>
             {
                 // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
                 (res as any).startTime = (res.config as any).startTime;
                 // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
                 (res as any).finalTime = process.hrtime.bigint();
                 return res;
-            })
+            },
+                res => // triggered when status code is not 2XX
```

Comment:
I don't think I understand the significance of what you changed here. These blocks of code look the same?

Reply:
It's wrapped in a try block now.
Axios has a client system but fetch does not.
This creates the client if it can but if Axios fails unexpectedly (like in the linked invalid url bug) we dont make a client BC we can't.

Then instead we will use fetch and record the error diagnostics for why the axios client failed. We also do this fetch request as a backup if axios doesnt work but has a client.